### PR TITLE
improve database retrieval and add icon/cover

### PR DESCRIPTION
**Pull request by johguentner**
improve database retrieval and add icon/cover:
- replace notion-api-docs link with updated page (Databases)
- add icon, iconType, cover, coverType to Database and Page with according extraction of different cover- and icon-types
- add database-notion-url retrieval to database
- add raw-information-retrieval to database-retrieval (add as empty properties with according information to property-key and options)
- add retrieval of available options to MultiSelect and Select during database retrieval
- don't retrieve rollup and formula if type-value is not given
- don't retrieve created_time and last_edited_time if retrieved content is null
- fix missing id's and wrong options-field (array in array) within response-json-files
- mark databases->all() as deprecated

### CODE REVIEW

File: src/Entities/Properties/Rollup.php [hunk 1]
```diff
@@ -22,22 +22,25 @@ protected function fillFromRaw(): void
...
-                break;
-            case 'array':
-                $this->setRollupContentArray();
-                break;
-            case 'date':
-                $this->setRollupContentDate();
-                break;
-            default:
-                throw new HandlingException("Unexpected rollupType {$this->rollupType}");
+        if(array_key_exists('type', $this->rawContent))
```
Comment (mechelon) [inline src/Entities/Properties/Rollup.php:25]: Arr::exists?

**Review by mechelon — CHANGES_REQUESTED** [review]

File: src/Entities/Properties/Rollup.php (hunk 1 above)
Comment (johguentner) [reply src/Entities/Properties/Rollup.php:25]: ✔️ Done

File: src/Entities/Properties/MultiSelect.php [hunk 2]
```diff
@@ -51,8 +56,16 @@ protected function fillFromRaw(): void
 
         $itemCollection = new Collection();
 
-        foreach ($this->rawContent as $item) {
-            $itemCollection->add(new SelectItem($item));
+        if(array_key_exists('options', $this->rawContent)){
```
[the remark this answers is not in the document]
Comment (johguentner) [reply src/Entities/Properties/MultiSelect.php]: ✔️ Done

File: src/Entities/Properties/LastEditedTime.php [hunk 3]
```diff
@@ -18,12 +18,11 @@ class LastEditedTime extends Property
     protected function fillFromRaw(): void
     {
         parent::fillFromRaw();
-        if ($this->rawContent == null) {
-            throw HandlingException::instance('The property-type is last_edited_time, however the raw data-structure is null. Please check the raw response-data.');
-        }
 
         try {
-            $this->content = new DateTime($this->rawContent);
+            if ($this->rawContent != null) {
```
[the remark this answers is not in the document]
Comment (johguentner) [reply src/Entities/Properties/LastEditedTime.php]: ✔️ Done

File: src/Entities/Properties/Formula.php [hunk 4]
```diff
@@ -20,14 +20,16 @@ protected function fillFromRaw(): void
...
-        if ($this->formulaType == 'string' || $this->formulaType == 'number' || $this->formulaType == 'boolean') {
-            $this->content = $this->rawContent[$this->formulaType];
-        } else if ($this->formulaType == 'date') {
-            $this->content = new RichDate();
-            if (isset($this->rawContent[$this->formulaType]['start'])) $this->content->setStart(new DateTime($this->rawContent[$this->formulaType]['start']));
-            if (isset($this->rawContent[$this->formulaType]['end'])) $this->content->setEnd(new DateTime($this->rawContent[$this->formulaType]['end']));
+        if (array_key_exists('type', $this->rawContent)) {
+            $this->formulaType = $this->rawContent['type'];
+
+            if ($this->formulaType == 'string' || $this->formulaType == 'number' || $this->formulaType == 'boolean') {
```
[the remark this answers is not in the document]
Comment (johguentner) [reply src/Entities/Properties/Formula.php]: ✔️ Done

File: src/Entities/Properties/Formula.php [hunk 5]
```diff
@@ -20,14 +20,16 @@ protected function fillFromRaw(): void
...
 
-        $this->formulaType = $this->rawContent['type'];
-
-        if ($this->formulaType == 'string' || $this->formulaType == 'number' || $this->formulaType == 'boolean') {
-            $this->content = $this->rawContent[$this->formulaType];
-        } else if ($this->formulaType == 'date') {
-            $this->content = new RichDate();
-            if (isset($this->rawContent[$this->formulaType]['start'])) $this->content->setStart(new DateTime($this->rawContent[$this->formulaType]['start']));
-            if (isset($this->rawContent[$this->formulaType]['end'])) $this->content->setEnd(new DateTime($this->rawContent[$this->formulaType]['end']));
+        if (array_key_exists('type', $this->rawContent)) {
```
[the remark this answers is not in the document]
Comment (johguentner) [reply src/Entities/Properties/Formula.php]: ✔️ Done

File: src/Entities/Database.php [hunk 6]
```diff
@@ -49,36 +125,150 @@ private function fillTitle(): void
...
+
+    /**
+     * @return string
+     */
+    public function getCoverType(): string
+    {
+        return $this->coverType;
+    }
+
+    public function getProperties(): Collection
```
[the remark this answers is not in the document]
Comment (johguentner) [reply src/Entities/Database.php:272]: ✔️ Done

**Review by mechelon — APPROVED** [review]

**Pull request merged**